tsdb/store.go

```diff
@@ -957,6 +946,20 @@ func IsRetryable(err error) bool {
 	return true
 }
 
+// DecodeStorePath extracts the database and retention policy names
+// from a given shard or WAL path.
+func DecodeStorePath(shardOrWALPath string) (database, retentionPolicy string) {
```

Comment:
The idea I had behind `DecodeStorePath` (`DecodeThingy`) is that it describes how to go from a path to `db`, `rp`, and potentially `sID`. If we change how we store any of these elements then this function and the alternative `EncodeStorePath` should be modified to make that happen.

I guess the issue here is that you haven't implemented the encode version of this function so I can understand your point about being defensive (someone might update how we store dbs/rps and not update this function).

Either way, I can't think of a case where your current approach wouldn't also break if things were changed.